xarray/tests/test_backends.py

```diff
@@ -1156,6 +1157,25 @@ def test_mask_and_scale(self):
                 expected = create_masked_and_scaled_data()
                 assert_identical(expected, ds)
 
+    def test_mask_and_scale_with_float64_scale_factor(self):
```

Comment:
It would be nice to test all of the edge cases just using `CFScaleOffsetCoder`, e.g.,
- what happens if `add_offset` and `scale_factor` have different types?
- what if they're integers instead of floats?

See the tests in `tests/test_coding.py` for examples.